sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseEndpointResolver.cs

```diff
@@ -73,6 +74,10 @@ public virtual void ProcessRequestHandlers(IExecutionContext executionContext)
                 };
                 requestContext.Request.Endpoint = uriBuilder.Uri;
             }
+            if (!string.IsNullOrEmpty(requestContext.ClientConfig.ServiceURL))
+            {
+                executionContext.RequestContext.UserAgentDetails.AddFeature(UserAgentFeatureId.ENDPOINT_OVERRIDE);
```

Comment:
Nit: `requestContext` is already defined, you don't need to use `executionContext.RequestContext` here.